court/tests.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+from court.models import Court
+from decimal import Decimal
+from django.db.utils import IntegrityError
+
+
+@pytest.mark.django_db
+class TestCourtModel:
+
+    @pytest.fixture
+    def court(self):
+        court = Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                                     city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+        return court
+
+    def test_validate_instance(self, court):
+        court_in_db = Court.objects.get(pk=court.pk)
+        assert court_in_db == court
+
+    def test_create_court_with_existing_id(self, court):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                                 city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+
+    def test_create_valid_court(self):
+        Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                             city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+
+    def test_create_court_with_negative_max_players(self):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(
+                courtID=1,
+                x=Decimal('11'),
+                y=Decimal('22'),
+                city='TEST_CITY',
+                neighborhood='TEST_NEIGHBORHOOD',
+                max_players=-100
+            )
+
+    def test_update_court(self, court):
+        new_court = court
+        assert new_court.max_players != 20
+        new_court.max_players = 20
+        new_court.save()
+        updated_court = Court.objects.get(pk=new_court.pk)
+        assert updated_court.max_players == 20
+
+    def test_delete_court(self, court):
+        test_court = court
+
+        # Check that test_court was indeed in Court.objects before deleting
+        assert test_court.courtID == Court.objects.get(pk=test_court.pk).courtID
+
+        test_court.delete()
+        with pytest.raises(Court.DoesNotExist):
+            Court.objects.get(pk=court.pk)
+
+    # Test: Check if can create new court with missing fields
+    def test_court_without_x(self):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(
+                courtID=1,
+                y=Decimal('1'),
+                city='TEST_CITY',
+                neighborhood='TEST_NEIGHBORHOOD',
+                max_players=10
+            )
+
+    def test_court_without_y(self):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(
+                courtID=1,
+                x=Decimal('1'),
+                city='TEST_CITY',
+                neighborhood='TEST_NEIGHBORHOOD',
+                max_players=10
+            )
+
+    def test_court_without_city(self, court):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(
+                courtID=1,
```

Comment:
I think the error here is raised because your trying to create a court with existing courtID, not because of a blank city field

Reply:
After changing the courtID to 800 (which doesnt exist if I'm not mistkaen) the tests remains the same - Integrity Error is raised.

Reply:
The best approach will be to have a positive test case that works, and then negative test cases, each only changes one field comparing to the one that works. This way you know for sure why you failed.

Reply:
Another thing, I don't think you want  users to define their court_ID (how would they know if it's valid?), it should be assigned automatically with a legal value (which will happen if you use Django defaults PK)